crypto/sleep.c

```diff
@@ -31,7 +31,8 @@ void OSSL_sleep(uint64_t millis)
     unsigned int s = (unsigned int)(millis / 1000);
     unsigned int us = (unsigned int)((millis % 1000) * 1000);
 
-    sleep(s);
+    if (s > 0)
+        sleep(s);
     usleep(us);
```

Comment:
Should `us` also be checked?
```suggestion
    if (us > 0)
        usleep(us);
```

Reply:
Prior discussions suggested that usleep is mandatory. It must be called to have any chance of a sched_yield() if OSSL_sleep(0) is called.